### PR TITLE
[samples] Server flow control example

### PR DESCRIPTION
**Pull request by eugeneo**

### CODE REVIEW

File: examples/cpp/flow_control/server_flow_control_client.cc [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  greeter->async()->SayHelloStreamReply(&ctx, &req, &reader);
+  reader.Start();
+  auto status = reader.WaitForDone();
+  if (status.ok()) {
+    std::cout << "Success\n";
+  } else {
+    std::cerr << "Failed with error: " << status.error_message() << "\n";
+  }
+  return 0;
+}
```
Comment (yashykt) [inline examples/cpp/flow_control/server_flow_control_client.cc:107]: add new line please

File: examples/cpp/flow_control/server_flow_control_server.cc [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                         absl::GetFlag(FLAGS_to_send));
+  ServerBuilder builder;
+  builder.RegisterService(&service);
+  builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());
+  // builder.RegisterService(&service);
+  std::unique_ptr<Server> server(builder.BuildAndStart());
+  std::cout << "Server listening on " << server_address << std::endl;
+  server->Wait();
+  return 0;
+}
```
Comment (yashykt) [inline examples/cpp/flow_control/server_flow_control_server.cc:131]: add new line please

File: examples/cpp/flow_control/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# Server flow control
+
+In server case, gRPC will pause the server implementation that is sending the
+messages too fast. Server implementation is in
+[server_flow_control_server.cc](server_flow_control_server.cc). It will write
+a specified number of responses of a specified size as fast as possible.
+As client-side buffer is filled, the write operation will block until the buffer
+is freed.
+
+A client implementation in [server_flow_control_server.cc](server_flow_control_server.cc)
```
Comment (yashykt) [inline examples/cpp/flow_control/README.md:21]: s/server_flow_control_server/server_flow_control_client

File: examples/cpp/flow_control/server_flow_control_client.cc [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+#include <grpcpp/grpcpp.h>
+
+#ifdef BAZEL_BUILD
+#include "examples/protos/helloworld.grpc.pb.h"
+#else
+#include "helloworld.grpc.pb.h"
+#endif
+
+ABSL_FLAG(std::string, target, "localhost:50051", "Server address");
+ABSL_FLAG(bool, bdp_probe, false,
```
Comment (yashykt) [inline examples/cpp/flow_control/server_flow_control_client.cc:36]: bdp increases the max frame size.. it shouldn't affect flow control for a single stream

**Review by yashykt — COMMENTED** [review]
I'm not sure how this works. Let's talk about it.

File: examples/cpp/flow_control/server_flow_control_client.cc (hunk 1 above)
Comment (eugeneo) [reply examples/cpp/flow_control/server_flow_control_client.cc]: Done

File: examples/cpp/flow_control/server_flow_control_server.cc (hunk 2 above)
Comment (eugeneo) [reply examples/cpp/flow_control/server_flow_control_server.cc]: Done

File: examples/cpp/flow_control/README.md (hunk 3 above)
Comment (eugeneo) [reply examples/cpp/flow_control/README.md]: Done

**Review by eugeneo — COMMENTED** [review]
Thanks for the review. I addressed what I could, need to discuss the rest in person.

File: examples/cpp/flow_control/server_flow_control_client.cc (hunk 4 above)
Comment (yashykt) [reply examples/cpp/flow_control/server_flow_control_client.cc:36]: My suggestion is to keep bdp enabled and set resource quota to something like 100MB, and then send 20MB size messages.. or something like that.

Comment (eugeneo) [reply examples/cpp/flow_control/server_flow_control_client.cc]: Done. Please take another look

**Review by eugeneo — COMMENTED** [review]
Addressed the remaining comment. Please take another look.

File: examples/cpp/flow_control/server_flow_control_server.cc [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  std::string server_address =
+      absl::StrCat("0.0.0.0:", absl::GetFlag(FLAGS_port));
+  grpc::EnableDefaultHealthCheckService(true);
+  grpc::reflection::InitProtoReflectionServerBuilderPlugin();
+  GreeterService service(absl::GetFlag(FLAGS_message_size),
+                         absl::GetFlag(FLAGS_to_send));
+  ServerBuilder builder;
+  builder.RegisterService(&service);
+  builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());
+  // builder.RegisterService(&service);
```
Comment (yashykt) [inline examples/cpp/flow_control/server_flow_control_server.cc:127]: please remove

File: examples/cpp/flow_control/server_flow_control_server.cc [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+using grpc::CallbackServerContext;
+using grpc::Server;
+using grpc::ServerBuilder;
+using grpc::ServerUnaryReactor;
+using grpc::Status;
+
+namespace {
+
+/**
```
Comment (yashykt) [inline examples/cpp/flow_control/server_flow_control_server.cc:52]: I think we are preferring to use `//` for comments.. Style guide says to prefer one or the other and be consistent.

**Review by yashykt — APPROVED** [review]

**Pull request closed without merging**